tests/beta/interfaces/test_qnode_autograd.py

```diff
@@ -69,7 +69,7 @@ def test_execution_with_interface(self, dev_name, diff_method):
         @qnode(dev, interface="autograd")
         def circuit(a):
             qml.RY(a, wires=0)
-            qml.RX(np.array(0.2, requires_grad=False), wires=0)
+            qml.RX(0.2, wires=0)

```

Comment:
This example explains why

* the Autograd tape was changed to assume non-differentiable by default, and
* The QNode was changed to treat its positional arguments as differentiable by default.

This allows us to match the existing behaviour of the old QNode, where positional arguments were wrapped as `Variable` objects if `requires_grad != False`.

Here, the RX gate has a constant parameter _inside_ the QNode -- the tape treats this as non-differentiable. If we were to instead do

```python
@qnode(dev, interface="autograd")
def circuit(a, b):
    qml.RY(a, wires=0)
    qml.RX(b, wires=0)
    return expval(qml.PauliZ(0))
```
and we called `circuit(0.1, 0.2)`, then both rotation gates will now be trainable.

In effect, we are imitating the behaviour where the presence of a parameter in the signature makes that parameter differentiable, which will then propagate down to gate arguments.

**Note:** the above discussion applies _only_ to the autograd interface. The TF and Torch interfaces continue to work the same as always.